network/consensus/base.go

```diff
@@ -24,28 +24,66 @@ import (
 	"github.com/insolar/insolar/log"
 )
 
-type exchangeResults map[core.RecordRef][]*core.ActiveNode
+// exchangeResults is thread safe results struct
+type exchangeResults struct {
+	mutex *sync.RWMutex
```

Comment:
Why are you using RWMutex if you use only Lock/Unlock?

Reply:
fixed